crates/bitcoin-da/src/service.rs

```diff
@@ -588,6 +594,114 @@ impl DaService for BitcoinService {
+    async fn extract_relevant_proofs(
+        &self,
+        block: &Self::FilteredBlock,
+        prover_pk: &[u8],
+    ) -> anyhow::Result<Vec<DaDataLightClient>> {
+        let mut completes = Vec::new();
+        let mut aggregate_idxs = Vec::new();
+
+        for tx in &block.txdata {
+            if !tx
+                .compute_wtxid()
+                .to_byte_array()
+                .as_slice()
+                .starts_with(&self.reveal_light_client_prefix)
+            {
+                continue;
+            }
+
+            if let Ok(parsed) = parse_light_client_transaction(tx) {
+                let tx_id = tx.compute_txid();
+                match parsed {
+                    ParsedLightClientTransaction::Complete(complete) => {
+                        if complete.public_key() != prover_pk
+                            && complete.get_sig_verified_hash().is_some()
+                        {
+                            // push only when signature is correct
+                            completes.push((tx_id, complete.body));
+                        }
+                    }
+                    ParsedLightClientTransaction::Aggregate(aggregate) => {
+                        if aggregate.public_key() != prover_pk
+                            && aggregate.get_sig_verified_hash().is_some()
+                        {
+                            // push only when signature is correct
+                            // collect tx ids
+                            aggregate_idxs.push((tx_id, aggregate));
+                        }
+                    }
+                    ParsedLightClientTransaction::Chunk(_chunk) => {
+                        // we ignore them for now
+                    }
+                }
+            }
+        }
+
+        // collect aggregated txs from chunks
+        let mut aggregates = Vec::new();
+        for (tx_id, aggregate) in &aggregate_idxs {
+            let mut body = Vec::new();
+            let Ok(chunk_ids) = aggregate.txids() else {
+                anyhow::bail!("{}: Failed to get txids from aggregate", tx_id);
+            };
+            if chunk_ids.is_empty() {
+                anyhow::bail!("{}: Empty aggregate tx list", tx_id);
+            }
+            for chunk_id in chunk_ids {
+                let tx_res = {
+                    let exponential_backoff = ExponentialBackoff::default();
+                    retry_backoff(exponential_backoff, || async move {
+                        self.client
+                            .get_transaction(&chunk_id, None)
+                            .await
+                            .map_err(|e| {
+                                use bitcoincore_rpc::Error;
+                                match e {
+                                    Error::Io(_) => backoff::Error::transient(e),
+                                    _ => backoff::Error::permanent(e),
+                                }
+                            })
```

Comment:
since we do

chunk - chunk - aggregate, we can assume we'll be able to query chunks when we see the aggregate.

so, if we get successfull response from rpc and the chunk tx does not exist -- we ignore the aggreagete (and all chunks we were able to retreive)